src/services/resource.js

```diff
@@ -1325,13 +1326,40 @@ module.exports = class resourceHelper {
 	 * @param {Integer} pageSize -  Used to limit the data. Used for pagination . If value is not passed, by default it will be 100
 	 * @returns {Object} - Response contain object of resources
 	 */
-	static async browseExistingList(organization_id, resourceIds = [], query, searchText = '', pageNo, pageSize) {
+	static async browseExistingList(
+		organization_id,
+		userRoles,
+		resourceIds = [],
+		query,
+		searchText = '',
+		pageNo,
+		pageSize
+	) {
 		try {
 			let result = {
 				data: [],
 				count: 0,
 			}
-			const resourceType = query[common.TYPE] ? query[common.TYPE].split(',') : ''
+			const roleTitles = userRoles.map((roles) => roles.title)
+			const rolePermissionDetails = await rolePermissionMappingQueries.findAll(
```

Comment:
@adithyadinesh0412  check rolePermissionDetails is empty or not, if empty send empty array in response